components/digital_ocean/actions/create-droplet/create-droplet.mjs

```diff
@@ -1,102 +1,128 @@
-      optional: true,
-    },
-    backups: {
-      type: "boolean",
-      description: "A boolean indicating whether automated backups should be enabled for the Droplet. Automated backups can only be enabled when the Droplet is created.",
-      optional: true,
-    },
-    ipv6: {
-      type: "boolean",
-      description: "A boolean indicating whether IPv6 is enabled on the Droplet.",
-      optional: true,
-    },
-    user_data: {
-      type: "string",
-      description: "A string containing 'user data' which may be used to configure the Droplet on first boot, often a 'cloud-config' file or Bash script. It must be plain text and may not exceed 64 KiB in size.",
-      optional: true,
-    },
-    private_networking: {
-      type: "boolean",
-      description: "A boolean indicating whether private networking is enabled for the Droplet. Private networking is currently only available in certain regions.",
-      optional: true,
-    },
-    volumes: {
-      type: "any",
-      description: "A flat array including the unique string identifier for each Block Storage volume to be attached to the Droplet. At the moment a volume can only be attached to a single Droplet.",
-      optional: true,
-    },
-    tags: {
-      type: "any",
-      description: "A flat array of tag names as strings to apply to the Droplet after it is created. Tag names can either be existing or new tags.",
-      optional: true,
-    },
-    monitoring: {
-      type: "boolean",
-      description: "A boolean indicating whether to install the DigitalOcean agent for monitoring.",
-      optional: true,
+      reloadProps: true,
+      async options() {
+        return await this.digitalOceanApp.fetchRegionsOpts();
+      },
     },
   },
-  async run({ $ }) {
-    var DigitalOcean = doWrapperModule.default,
-      api = new DigitalOcean(this.digital_ocean.$auth.oauth_access_token, this.page_size);
-
-    try {
-      var data = {
-        name: this.name,
-        region: this.region,
-        size: this.size,
-        image: this.image,
-        ssh_keys: this.ssh_keys,
-        backups: this.backups,
-        ipv6: this.ipv6,
-        user_data: this.user_data,
-        private_networking: this.private_networking,
-        volumes: this.volumes,
-        tags: this.tags,
-        monitoring: this.monitoring,
+  async additionalProps() {
+    let dynamicProps = {};
+    if (this.region) {
+      dynamicProps = {
+        ...dynamicProps,
+        image: {
+          label: "Image",
+          type: "string",
+          description: "The image ID of a public or private image, or the unique slug identifier for a public image. This image will be the base image for this Droplet.",
+          reloadProps: false,
+          options: async () => {
+            return await this.digitalOceanApp.fetchImageOpts(this.region);
+          },
```

Comment:
Did you test this already? because I remember having this discussion in slack [here](https://pipedream-users.slack.com/archives/CPTJYRY5A/p1648568306665749) so I wonder if this works for you